fixed an issue with float32 and added update count to update/insert/delete rows

kujukuju opened the pull request:
NOTE...

I added the return count after the success: bool value. This seems to go against your convention in this library, but I think there are quite a few cases where you would just want to know if the query succeeded and don't care how many rows were updated.

Separately if I added the count first it would break all current code, for better or worse.

Also the float32 code seemed to actually have a bug. It was directly casting into float32 while the data being returned from postgres was float64. This was actually returning the wrong value.

I'm not sure if you want to accept the first change or not, so this probably should have been two commits, but oh well. Edit as you want.

kujukuju committed:
fixed float32 issue, added update count to executed statements without return types

rluba commented:
Thanks! Could you add the float32 case to the example so that we have a "test" for it?

rluba commented:
>  This seems to go against your convention in this library, but I think there are quite a few cases where you would just want to know if the query succeeded and don't care how many rows were updated.


No, you’re right. It should be the second argument for exactly this reason.

rluba commented on module.jai:
```diff
@@ -642,6 +647,7 @@ str_from_numeric :: (data: *u8, len: int, row: int, col: int) -> string, success
 
 #import "Basic";
 #import "Pool";
+#import "String";
```
Why is this necessary now?

kujukuju replied:
Just for parse_int for:
```jai
            rows_affected := to_string(PQcmdTuples(query_res));
            rows_affected_count := parse_int(*rows_affected);
```

If there's another way to parse I'll swap it over and remove this.

rluba replied:
You can just use `string_to_int` from `Basic`.

rluba commented on module.jai:
```diff
@@ -183,26 +185,29 @@ get_last_query_result :: (conn: *PGconn) -> *PGresult {
 }
 
-check_query_result :: (query_res: *PGresult) -> has_results: bool, success: bool {
+check_query_result :: (query_res: *PGresult) -> has_results: bool, updated_rows: int, success: bool {
```
`update_rows` should be the third return value here for the same reason as why you made it the second value for `execute`

kujukuju committed:
reordered count and success, added float testcase

kujukuju committed:
added support for actual float4 too

kujukuju commented:
So actually @rluba the original bug was trying to put a float8 into a struct's float4. It seems the default value for FLOAT in postgres is float8.

This is the bug I fixed, so I added a test case for this.

I also just added float4 support which is basically just identical to float8, and I added the same 2 test cases for float4.

kujukuju committed:
whoops, I removed example.jai

rluba commented:
I’ve manually cleaned this up and merged it into `master`.
